events/events.go

```diff
@@ -0,0 +1,197 @@
+package events
+
+import (
+	"errors"
+	"sync"
+)
+
+var errClosed = errors.New("publisher is closed")
+
+// GetFn returns the event
+type GetFn func() interface{}
+
+// Event is an event
+type Event interface {
+	// OnGet sets the function to be executed when Get() is called
+	OnGet(GetFn)
+	// Get the latest Event
+	Get() interface{}
+}
+
+// Publisher is a publisher for an event
+type Publisher interface {
+	Event
```

Comment:
It's kind of strange that `Publisher` extends an `Event` perhaps a better name for `Event` would be `EventDataSource`?